Marlin/src/module/tool_change.cpp

```diff
@@ -388,10 +385,17 @@ void tool_change(const uint8_t tmp_extruder, const float fr_mm_s/*=0.0*/, bool n
 
         #if ENABLED(DUAL_X_CARRIAGE)
 
+          // Don't move the new extruder out of bounds
+          if (tmp_extruder
+            ? !WITHIN(current_position[X_AXIS], X2_MIN_POS, X2_MAX_POS)
+            : !WITHIN(current_position[X_AXIS], X1_MIN_POS, X1_MAX_POS)) no_move = true;
```

Comment:
`X2_MAX_POS` is defined as the distance between the extruders when both are homed. It is not the same as the maximum position unless the `X_HOME_POS` (and `X1_HOME_POS`) are both zero. So the `X2_MAX_POS` here should probably be `X1_MIN_POS + X2_MAX_POS`.